test/e2e/nftables_test.go

```diff
@@ -69,15 +94,24 @@ var _ = Describe("Nftables", func() {
 		// Wait for all goroutines to finish
 		err = g.Wait()
 		Expect(err).ToNot(HaveOccurred())
-		nodeName := nodeList.Items[0].Name
 
-		By("Rebooting first node: " + nodeName + "and waiting for disconnect")
+		fmt.Printf("Waiting for %s after applying MachineConfiguration...\n", waitDuration)
+		time.Sleep(waitDuration)
+		waitForMCPReady(cs, mcTimeout)
```

Comment:
So firewall.Apply creates a MC and the test is waiting for it? If I'm not familiar with I won't be able to understand why suddenly there is a wait for a MC
Maybe better create the MC in firewall but apply it in the test so we know what we are waiting for

Reply:
so we are creating the MC (applying the firewall) and waiting for that to be ready
u think maybe i can change the apply function to be applyAndWait for mcp (not sure about that)?